extension/httpfs/s3fs.cpp

```diff
@@ -193,7 +193,13 @@ S3AuthParams S3AuthParams::ReadFrom(FileOpener *opener, FileOpenerInfo &info) {
 	if (FileOpener::TryGetCurrentSetting(opener, "s3_endpoint", value, info)) {
 		endpoint = value.ToString();
 	} else {
-		endpoint = "s3.amazonaws.com";
+		if (StringUtil::StartsWith(info.file_path, "gcs://")) {
```

Comment:
The idea of this change is that when a user queries somethings like:
```
SELECT * FROM 'gcs://test-bucket/some/file';
```
We can automatically set the endpoint as you are doing here. However I think this codepath is currently never hit, because the TryGetCurrentSetting now always returns true, setting the endpoint to an empty string in this case. Could you rewrite this in such a way that when the `s3_endpoint` does not exist or is empty, we set the default based on the prefix?